### ISSUE TITLE
YAD 7.3: Dialog height too large

### ISSUE THREAD

**Issue by sarether** [v1cont/yad#107]
Dialog height isn't minimum unless specify a height or specify a small value for the height.
But any width is allowed and that's OK, as with previous YAD versions.

```
yad --title="app - title" --width=450 --text="text text text text text text text text text text text text text text text text.
text text text text?" --on-top --center --image=gtk-dialog-question --window-icon=$ICON --button="yad-no:1" --button="yad-yes:0"
```
YAD 7.2 and older versions:
![PrtSc-21 10 20_Wednesday_19:57:11 61](https://user-images.githubusercontent.com/17093556/96753491-7787c580-13bf-11eb-94b8-286a1419429b.png)

YAD 7.3:
![PrtSc-21 10 20_Wednesday_19:56:20 51](https://user-images.githubusercontent.com/17093556/96753506-7c4c7980-13bf-11eb-9511-fb17b1e0f0b8.png)

**Comment by buc-me** [comment]
There are two possible solution on the old site: https://sourceforge.net/p/yad-dialog/tickets/322/#f7e3/5aa0

**Comment by manuel-192** [comment]
I can confirm the dialog height issue. As said, in 7.2 it was OK, but in 7.3 it is not.

**Comment by manuel-192** [comment]
Looks like commit 8442c52a2048e85e410eb8945641222eb0b9dc34 into main.c causes this issue. If that commit is reverted, then the size is as expected.

Note that I didn't test it thoroughly, only with one simple example.

**Comment by eworm-de** [comment]
I can confirm this issue as well, commit 8442c52a2048e85e410eb8945641222eb0b9dc34 is bad.

**Comment by buc-me** [comment]
Hi all,

Could you please check whether the patch (proposed on the old site) fixes your issues?
https://sourceforge.net/p/yad-dialog/tickets/_discuss/thread/21ece8aec1/f7e3/5aa0/attachment/yad-7.3-size-request.patch

**Comment by eworm-de** [comment]
Yes, works for me.
What exactly was commit 8442c52a2048e85e410eb8945641222eb0b9dc34 supposed to fix?

**Comment by buc-me** [comment]
I believe this one: https://sourceforge.net/p/yad-dialog/tickets/322/

**Comment by eworm-de** [comment]
Any news on this one? Is there a reason this is not committed?

**Comment by eworm-de** [comment]
Still an issue in version 8.0...

**Comment by sarether** [comment]
> Still an issue in version 8.0...

Yes,

**Comment by killajoe** [comment]
So what can we do to get the patch merged? it is working and we do not see other issues with usage of patched version:
![EndeavourOS welcome YAD](https://forum.endeavouros.com/uploads/default/original/3X/1/8/189ec7f1eed9a1bc20d55de046f59c58afa37b8c.jpeg)
At EndeavourOS ISO we would like to see this solved as it is confusing users a lot and simply looking screwed ;)

@v1cont 

Tested with latest Version and the pach:
https://github.com/killajoe/yad-patched

**Comment by v1cont** [comment]
please use --text-width option with long strings in dialog's text

**Comment by eworm-de** [comment]
Using `--text-width` results in just reducing the issue or unnecessarily wide windows. But the results are not perfect and this is far from intuitive.

What's wrong with applying the changes?

**Comment by buc-me** [comment]
Hi all,

Yet another patch which should fix the issue and the same time allow shrinking of the normal dialogs, see on the old site: https://sourceforge.net/p/yad-dialog/tickets/322/#6f2d

**Comment by buc-me** [comment]
Please, test whether this new patch resolves your issue(s),
https://sourceforge.net/p/yad-dialog/tickets/_discuss/thread/21ece8aec1/6f2d/attachment/yad-github-issue-107.patch

**Comment by killajoe** [comment]
may a stupid question... but why there is "on the old site" at Sourceforge where  Last Update: 2017-11-18  and latest version of YAD is: yad-0.40.0.tar.xz while here at GitHub we have: yad-9.1.tar.xz ...  then i see a message with a "new" patch for version 8.0 ... 
Very confusing ;)

**Comment by buc-me** [comment]
Johannes Kamprad wrote:
>
> may a stupid question... but why there is "on the old site" at 
> Sourceforge where Last Update: 2017-11-18 and latest version of YAD 
> is: yad-0.40.0.tar.xz while here at GitHub we have: yad-9.1.tar.xz ... 
> then i see a message with a "new" patch for version 8.0 ...
> Very confusing ;)
>
In short: just habits. Someone starts years ago, and still feel more 
comfortable with SF, especially for occasional participation.

Plus such a people often use "oldschool" browsers (palemoon, seamonkey), 
which github dislikes (without any objective reasons for this).

For clarity, a counter "stupid" question :) -- Why you use shell+yad, 
whereas the use of, say, python for the same purpose is much more power? 
And it seems so easy for modern advanced developer to learn python... 
Answering this, you will answer your initial question too. ;)
</offtopick>

**Comment by killajoe** [comment]
Sorry for asking, but i am still confused about the relation between GitHub and Sourceforge repositories and versions used. 
But i will be calm now. 
It was not my intention to step on your feet.

**Comment by buc-me** [comment]
Johannes Kamprad wrote:
>
>  i am still confused about the relation between GitHub and Sourceforge 
> repositories and versions used.
>

Well, the old SF repo seems just stalled on 0.40, without a proper 
mention that the project had been moved to github. Thus some 
distribution even are not aware of it and still continue to provide 0.40 
only. :(

Once seeing a mention of newer versions somewhere in SF bugs, I guessed 
that the project had moved to github.

Many projects, initially being on SF, now are moved to github or gitlab. 
Some of them completely forget about SF, but some still use SF for 
"traditional" maillists and bug tracking, and update both sites 
properly. So it is not abnormal to still use SF nowadays, which I prefer 
to do for some reasons too.

**Comment by mackendw** [comment]
same issue with Notepads...specifying the --height option doesn't affect the sizing of the window created.  tested version 7.3 and 9.1

**Comment by eworm-de** [comment]
> Please, test whether this new patch resolves your issue(s),
> https://sourceforge.net/p/yad-dialog/tickets/_discuss/thread/21ece8aec1/6f2d/attachment/yad-github-issue-107.patch

Yes, it does. With this applied the window is sized as expected.

**Comment by sarether** [comment]
Versions 9.3 and few early are OK.

**Comment by eworm-de** [comment]
This is not true... (But looks like you noticed yourself.)

With this command...
```
yad --info --title "$(yad --version)" --no-markup --text "$(ip addr show)" --button=Close
```
... the result looks still bad with `yad 10.1`:
![Screenshot_2021-11-24_09-54-33](https://user-images.githubusercontent.com/970853/143206211-3545689e-c772-42c4-9949-d00e313d5834.png)

**Comment by sarether** [comment]
> This is not true... (But looks like you noticed yourself.)
I forgot that I patched YAD.

**Comment by eworm-de** [comment]
Any upstream news on this?
Would be nice if I could drop my custom rebuild.

**Comment by eworm-de** [comment]
Still an issue with version 11.0... Any reason nothing happens here?

**Comment by eworm-de** [comment]
Sure I did see the news. And I am with you. All the best to you and the people in the Ukraine.

But this issue exists since October 2020 and has been present in a number of releases. Even patches are around. So this could have been fixed before Putin went mad (or at lease before he went *that* mad).

**Comment by magpie514** [comment]
@eworm-de 
I am really, really sorry, but that's a terrible thing to say to someone in a warzone. He clearly has more important things to worry about and it's not his fault.
YAD is an excellent piece of software and it's painful to see it affected by war, but right now I'm more concerned about the developer and his loved ones.

If things to back to normal, open a pull request since you said you have a custom build, that'll help fix things for everyone and take a load off the dev's back.

**Comment by v1cont** [comment]
it's all right. now i staying in reserve for territorial defense forces and waiting for rotation. so i have some free time. the goals of last release was primary political and ethical, not technical. main reason was drop the russian translation from yad. as for issue there must be an algorithm for calculating the text dimensions which i cannot implement right now. but i promise i'll implement it as soon as possible

**Comment by v1cont** [comment]
i will not apply this fix because it makes yad impossible to reduce it's width

feel free to patch yad by themselves, or use combination of --width, --text-width and --fixed options

**Comment by eworm-de** [comment]
There are at least two patches available. Both are not acceptable?
Can you give an example command that does not behave as expected with one of the patches applied?

Also I think I could not get a satisfying result with the properties given above. Can you give values that should work?

**Comment by Carusu** [comment]
Maybe my case is a limit test-case but I do always get too large windows, not matter what combo of `--width`, `--text-width` and `fixed` i try.

What I want is a notification of the 3 most memory intensive process and I want it to refresh. Here is my script:

`#!/bin/bash
export fpipe=/tmp/mem.yad03
mkfifo $fpipe
trap "rm -f $fpipe" EXIT
exec 3<> $fpipe

yad \
  --close-on-unfocus \
  --no-buttons \
  --undecorated \
  --info \
  --text-info \
  --cycle-read \
  --height=620 --width=220 \
  --text-width=220 \
  --fixed \
  --listen \
  < $fpipe &

yad_pid=$!
  
while [[ -d /proc/$yad_pid ]]; do
  echo -e '\f'
  ps -eo comm,pid,pmem --sort -pmem | head -4 
  sleep 1
done  > $fpipe`

It works but the sizes are always off
